nbeats_keras/model.py

```diff
@@ -1,54 +1,89 @@
 import numpy as np
+
 from keras import backend as K
-from keras.layers import Input, Dense, Lambda, Subtract, Add
+from keras.layers import Input, Dense, Lambda, Subtract, Add, Reshape
+from keras.layers import Concatenate
 from keras.models import Model
 from keras.optimizers import Adam
 
 
 class NBeatsNet:
-    SEASONALITY_BLOCK = 'seasonality'
-    TREND_BLOCK = 'trend'
     GENERIC_BLOCK = 'generic'
-
-    def __init__(self,
-                 stack_types=(TREND_BLOCK, SEASONALITY_BLOCK),
+    TREND_BLOCK = 'trend'
+    SEASONALITY_BLOCK = 'seasonality'
+    
+    def __init__(self, 
+                 input_dim, 
+                 exo_dim, 
+                 backcast_length, 
+                 forecast_length, 
+                 stack_types=[TREND_BLOCK, SEASONALITY_BLOCK],
                  nb_blocks_per_stack=3,
-                 forecast_length=2,
-                 backcast_length=10,
-                 thetas_dim=(4, 8),
+                 thetas_dim=[4, 8],
                  share_weights_in_stack=False,
-                 hidden_layer_units=256):
-        self.backcast_length = backcast_length
-        self.forecast_length = forecast_length
-        self.units = hidden_layer_units
+                 hidden_layer_units=256
+                 ):
+        
         self.stack_types = stack_types
         self.nb_blocks_per_stack = nb_blocks_per_stack
-        self.share_weights_in_stack = share_weights_in_stack
         self.thetas_dim = thetas_dim
-        self.best_perf = 100.0
+        self.units = hidden_layer_units
+        self.share_weights_in_stack = share_weights_in_stack
+        self.backcast_length = backcast_length
+        self.forecast_length = forecast_length
+        self.best_perf = np.inf
```

Comment:
`best_perf = np.inf` to be moved to the trainer.

Reply:
Updated